src/huggingface_hub/repocard_data.py

```diff
@@ -76,6 +81,38 @@ class EvalResult:
 
 @dataclass
 class CardData:
+    def __init__(self, **kwargs):
+        self.__dict__.update(kwargs)
+
+    def to_dict(self):
+        """Converts CardData to a dict.
+
+        Returns:
+            `dict`: CardData represented as a dictionary ready to be dumped to a YAML
+            block for inclusion in a README.md file.
+        """
+
+        data_dict = copy.deepcopy(self.__dict__)
+        self._to_dict(data_dict)
```

Comment:
As discussed via DM, I'm not a fan of hacking `_to_dict` to make this work. Simply implementing `to_dict` for each might be cleaner, but I have a feeling others might prefer the `_to_dict` approach (the current one). WDYT @LysandreJik?